element-hq/element-x-android
# Change a room's permissions power levels

## Type of change


## Content

- Bumps the Rust SDK to be able to use some new FFI APIs.
- Added a new 'permissions' section in the Roles and permissions screen to change the power level required to perform certain changes in the room.
- Opening any of these takes you to a new screen where you can modify these settings.
- Added a 'reset permissions' list item to return those values to the default ones according to the Matrix spec.

## Motivation and context

#2259.

## Screenshots / GIFs

In the PR. Some screenshots were renamed, since their preview function was renamed too.

## Tests

- Go to a room where you're and admin, and open its roles and permissions settings.
- Modify some power level in the permissions section
- Make sure users with a lower power level can't perform the action anymore.
- Reset the permissions.
- Check the reset permissions stand.

## Tested devices

- OS version(s): 14

## Checklist
